internal/exec/engine.go

```diff
@@ -92,12 +101,7 @@ func (e Engine) Run(stageName string) bool {
 
 // acquireConfig returns the configuration, first checking a local cache
 // before attempting to fetch it from the provider.
-func (e *Engine) acquireConfig() (cfg types.Config, f resource.Fetcher, err error) {
-	f, err = e.OEMConfig.NewFetcherFunc()(e.Logger)
-	if err != nil {
-		e.Logger.Crit("failed to generate fetcher: %s", err)
-		return
-	}
+func (e *Engine) acquireConfig() (cfg types.Config, err error) {
```

Comment:
Here and in other methods, perhaps we should check that both `e` and `e.fetcher` are non-nil before dereferencing them?

Reply:
If only golang had a way to allow mutation of a variable without having nil checks... I think it's reasonable to not check e (since this is only ever called as `e.funcname()` on a struct e instead of a pointer to the struct e), but e.fetcher is probably worth checking

Reply:
I  think is it reasonable as is then (I tend to over-sprinkle nil-checks across codebase).